R/lint.R

```diff
@@ -63,6 +63,32 @@ lint <- function(filename, linters = NULL, cache = FALSE, ..., parse_settings =
     names(linters) <- auto_names(linters)
   }
 
+  linters <- Map(
+    function(obj, name) {
+      if (inherits(obj, "linter")) {
+      } else if (inherits(obj, "function")) {
+        if (is.null(formals(obj))) {
```

Comment:
A word of caution with builtins (not really important here IMO), but
```
> formals(sin)
NULL
```
Safer way:
```
> formals(args(sin))
$x

```